salt/modules/dockermod.py

```diff
@@ -912,6 +913,11 @@ def compare_container(first, second, ignore=None):
                 if item == 'Links':
                     val1 = sorted(_scrub_links(val1, first))
                     val2 = sorted(_scrub_links(val2, second))
+                if item == 'Ulimits':
+                    if isinstance(val1, collections.Iterable):
+                        val1 = sorted(val1)
```

Comment:
`TypeError: unorderable types: dict() < dict()` in py3, so this needs to sort using the dict itself, maybe by `Name`, then `Hard`, then `Soft`.